lib/github_changelog_generator/generator/entry.rb

```diff
@@ -0,0 +1,309 @@
+require "github_changelog_generator/generator/section"
+
+module GitHubChangelogGenerator
+  class Entry
+    attr_reader :content
+
+    def initialize(options = {})
+      @content = ""
+      @options = options
+    end
+
+    # Generates log entry with header and body
+    #
+    # @param [Array] pull_requests List or PR's in new section
+    # @param [Array] issues List of issues in new section
+    # @param [String] newer_tag Name of the newer tag. Could be nil for `Unreleased` section
+    # @param [Hash, nil] older_tag Older tag, used for the links. Could be nil for last tag.
+    # @return [String] Ready and parsed section
+    def create_entry_for_tag(pull_requests, issues, newer_tag, older_tag = nil)
+      newer_tag_link, newer_tag_name, newer_tag_time = detect_link_tag_time(newer_tag)
+
+      github_site = @options[:github_site] || "https://github.com"
+      project_url = "#{github_site}/#{@options[:user]}/#{@options[:project]}"
+
+      # If the older tag is nil, go back in time from the latest tag and find
+      # the SHA for the first commit.
+      older_tag_name =
+        if older_tag.nil?
+          @fetcher.commits_before(newer_tag_time).last["sha"]
```

Comment:
Should an entry fetch it's preceding commits, or is this logic part of the larger changelog generation?

If the former, then the entry class needs access to the fetcher (pass through options?). If the latter, then this needs to move out of the entry class back to the generator class.

Reply:
The `create_entry_for_tag` method should not make calls out to github, so instead of having the fetcher passed through in the case of a nil older tag, the older tag name should be a required parameter and the generator class should either pass the older tag name or the first commit sha in the absence of an older tag.

The `older_tag_link` variable used in `generate_header` is misnamed since it is actually just a tag link, not a name. It is also possibly a sha and not a tag, but I'll ignore that for now.